src/settings_util.c

```diff
@@ -11,12 +11,44 @@
  */
 
 #include <stddef.h>
+#include <stdio.h>
 
 #include <libsettings/settings_util.h>
 
+int settings_format(const char *section,
+                    const char *name,
+                    const char *value,
+                    const char *type,
+                    char *buf,
+                    uint16_t blen)
+{
+  int n = 0;
+  int l = 0;
+
+  const char *tokens[SETTINGS_TOKENS_TYPE] = {section, name, value, type};
+
+  for (int i = 0; i < SETTINGS_TOKENS_TYPE; ++i) {
+    const char *token = tokens[i];
+
+    if (token == NULL) {
+      break;
+    }
+
+    l = snprintf(&buf[n], blen - n, "%s", token);
+
+    if ((l < 0) || (l >= blen - n)) {
+      return -1;
+    }
+
+    n += l + 1;
+  }
+
+  return n;
+}
+
 /* Parse SBP message payload into setting parameters */
 settings_tokens_t settings_parse(const char *buf,
-                                 uint8_t blen,
+                                 uint16_t blen,
```

Comment:
current limitation in this value is defined by [`libsbp/sbp.h#L48`](https://github.com/swift-nav/libsbp/blob/8ced9d3d6753390cdd0b53538322f070f52cfbe8/c/include/libsbp/sbp.h#L48). is increasing the size necessary?

edit: and looking around in general there are some `int` typed `blen` variables, so this seems inconsistent overall. maybe a typedef like `sbp_buffer_size_t` or something similar might help to constrain this properly?

Reply:
I'll try to come up with something unified. The reason for this change originated from the buffer sizes on sbp_setting_daemon being `256` which wraps around to 0 with `uint8_t`. It had me scratching my head for a while.

Reply:
https://swift-nav.atlassian.net/projects/ESD/issues/ESD-1028